sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/cli/Type.scala

```diff
@@ -0,0 +1,376 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.thriftserver.cli
```

Comment:
This is essentially copied from Hive Type.java, removing the types that we don't support, right?

Maybe instead of cloning it like that, still use the Hive types that Hive's GetTypeInfoOperation uses, and just have a filter like
```
val unsupportedTypes = Set("INTERVAL_YEAR_MONTH", "INTERVAL_DAY_TIME", "ARRAY", "MAP", "STRUCT", "UNIONTYPE", "USER_DEFINED")
...
if (!unsupportedTypes.contains(typeInfoGetName)) {
  val rowData = ...
}
```

There may be value in the future in cloning that, if our types would diverge more from Hive types, but I think for now we can just reuse it instead of cloning these several hundred lines of code.

Reply:
emmmm since for Hive.
In 1.2.1 Type class is `org.apache.hive.service.cli.Type`
In 2.3.x Type class is `org.apache.hadoop.hive.serde2.thrift.Type`

Spark only return it's DataType, I add a new Class(Type) for Spark.
If not we may need to add more method to ThriftServerShimUtils.

Reply:
I am working on make Spark's own ThriftServer, don't rely on hive version. 
In my subconscious，don't want to add method on ThriftserverShimUtils.

Reply:
If this only clones the Hive types, skipping the ones we don't support, I'd prefer adding it to the shim and not cloning the code.
I don't think Spark would further diverge from Hive types. If it happens, we could clone it at that point.
@wangyum what do you think?

Reply:
Not all same, such as `INTEGER`, Spark IntegerType's simpleStringName is INTEGER, in hive Types, there is INT. for `NULL`, change the name to `NULL`, in hive is `void`.

Reply:
Ok. That seems like a good enough argument to just clone it.
Please list these differences in the PR description.

Reply:
OK, I will add to description.